pkg/cmd/util/clientcmd/gating.go

```diff
@@ -1,64 +1,49 @@
 package clientcmd
 
 import (
-	"encoding/json"
 	"fmt"
 
-	"github.com/blang/semver"
-
 	"github.com/openshift/origin/pkg/client"
-	"github.com/openshift/origin/pkg/version"
 )
 
-// Gate returns an error if the server is below minServerVersion or above/equal maxServerVersion.
-// To test only for min or only max version, set the other string to the empty value.
-func Gate(ocClient *client.Client, minServerVersion, maxServerVersion string) error {
-	if len(minServerVersion) == 0 && len(maxServerVersion) == 0 {
-		return fmt.Errorf("No version info passed to gate command")
+// Return err if the server does not support any of the legacy policy objects (< 3.7)
+func LegacyPolicyResourceGate(ocClient *client.Client) error {
+	return ResourceGate(ocClient, "authorization.openshift.io/v1", []string{
+		"clusterpolicies",
+		"clusterpolicybindings",
+		"policies",
+		"policybindings",
+	})
+}
+
+// Do a discovery for server resources of the provided groupVersion, returning nil if any one of resourceNames is supported
+// by the server.
+func ResourceGate(ocClient *client.Client, groupVersion string, resourceNames []string) error {
```

Comment:
This should take a variatric list of GVR structs, not []string and a separate GV.